src/lib/react/ReactMacro.hx

```diff
@@ -136,42 +202,44 @@ class ReactMacro
 
 					var props = makeProps(spread, attrs, pos);
 
-					var args = [type, props].concat(children);
-					return macro react.React.createElement($a{args});
+					var args = [type, props].concat(children.individual);
+					macro @:pos(n.name.pos) react.React.createElement($a{args});
 				}
+			case CSplat(_): 
+				c.pos.error('jsx does not support child splats');
+			case CIf(cond, cons, alt): 
+				macro @:pos(cond.pos) if ($cond) ${body(cons)} else ${body(alt)};
+			case CFor(head, expr):
+				macro @:pos(head.pos) ([for ($head) ${body(expr)}]:Array<Dynamic>);
+			case CSwitch(target, cases):
+				ESwitch(target, [for (c in cases) {
+					guard: c.guard,
+					values: c.values,
+					expr: body(c.children)
+				}], null).at(target.pos);
+			default: c.pos.error('jsx does not support control structures');//already disabled at parser level anyway
 		}
-	}
-
-	static function parseJsxAttr(value:String, pos:Position)
-	{
-		var ast = JsxParser.parseText(value);
-		return switch (ast)
-		{
-			case JsxAst.Text(value):
-				return macro $v{value};
 
-			case JsxAst.Expr(value):
-				return Context.parse(value, pos);
-
-			default: null;
-		}
-	}
+	static function body(c:Children)
+		return macro ($a{children(c).individual}:Array<Dynamic>);
+	
+	static var componentsMap:Map<String, ComponentInfo> = new Map();
 
 	static function genLiteral(type:Expr, props:Expr, ref:Expr, key:Expr, pos:Position)
 	{
 		if (key == null) key = macro null;
 		if (ref == null) ref = macro null;
 
-		var fields = [
-			{field: "@$__hx__$$typeof", expr: macro untyped __js__("$$tre")},
+		var fields:Array<ObjectField> = [
+			{field: #if (haxe_ver < 4) "@$__hx__$$typeof" #else "$$typeof", quotes: DoubleQuotes #end, expr: macro untyped __js__("$$tre")},
```

Comment:
Shouldn't it be `quotes: Quoted` instead?
https://github.com/HaxeFoundation/haxe/commit/b6e5d6645dde34f58c56d6ec25fb66dcb089c176